universal-application-tool-0.0.1/app/services/export/JsonExporter.java

```diff
@@ -0,0 +1,148 @@
+  }
+
+  public String export(ProgramDefinition programDefinition) {
+    DocumentContext jsonApplications = makeEmptyJsonArray();
+    PaginationResult<Application> applicationPaginationResult =
+        programRepository.getApplicationsForAllProgramVersions(
+            programDefinition.id(), PaginationSpec.MAX_PAGE_SIZE_SPEC, Optional.empty());
+
+    for (Application application : applicationPaginationResult.getPageContents()) {
+      CfJsonDocumentContext applicationJson = buildJsonApplication(application, programDefinition);
+      jsonApplications.add("$", applicationJson.getDocumentContext().json());
+    }
+
+    return jsonApplications.jsonString();
+  }
+
+  private CfJsonDocumentContext buildJsonApplication(
+      Application application, ProgramDefinition programDefinition) {
+    ReadOnlyApplicantProgramService roApplicantProgramService =
+        applicantService.getReadOnlyApplicantProgramService(application, programDefinition);
+    CfJsonDocumentContext jsonApplication = new CfJsonDocumentContext(makeEmptyJsonObject());
+
+    jsonApplication.putString(
+        Path.create("program_name"), application.getProgram().getProgramDefinition().adminName());
+    jsonApplication.putLong(Path.create("program_version_id"), application.getProgram().id);
+    jsonApplication.putLong(Path.create("applicant_id"), application.getApplicant().id);
+    jsonApplication.putLong(Path.create("application_id"), application.id);
+    jsonApplication.putString(
+        Path.create("language"),
+        roApplicantProgramService.getApplicantData().preferredLocale().toLanguageTag());
+    jsonApplication.putString(Path.create("create_time"), application.getCreateTime().toString());
+    jsonApplication.putString(
+        Path.create("submitter_email"), application.getSubmitterEmail().orElse("Applicant"));
+
+    if (application.getSubmitTime() == null) {
+      jsonApplication.putNull(Path.create("submit_time"));
+    } else {
+      jsonApplication.putString(Path.create("submit_time"), application.getSubmitTime().toString());
+    }
+
+    for (AnswerData answerData : roApplicantProgramService.getSummaryData()) {
+      // Answers to enumerator questions should not be included because the path is incompatible
+      // with the JSON export schema. This is because enumerators store an identifier value for
+      // each repeated entity, which with the current export logic conflicts with the answers
+      // stored for repeated entities.
+      if (answerData.questionDefinition().isEnumerator()) {
+        continue;
+      }
+
+      switch (answerData.questionDefinition().getQuestionType()) {
+        case CHECKBOX:
+          {
+            MultiSelectQuestion multiSelectQuestion =
+                answerData.applicantQuestion().createMultiSelectQuestion();
+            Path path = multiSelectQuestion.getSelectionPath().asApplicationPath();
+
+            if (multiSelectQuestion.getSelectedOptionsValue().isPresent()) {
+              int i = 0;
```

Comment:
this isn't getting incremented.  Either fix that but or bake in the 0 in line.

Reply:
Also an indicator of missing testing if this is a bug.

Reply:
one would think! adding multiple multiselect answers to an array is covered in the unit test but apparently jsonpath is appending to the array despite the index getting passed in here. i'll look into the behavior a little more closely.